bson-kotlinx/src/main/kotlin/org/bson/codecs/kotlinx/BsonDecoder.kt

```diff
@@ -66,7 +66,7 @@ internal sealed interface BsonDecoder : Decoder, CompositeDecoder {
             }
         }
 
-        fun createBsonDecoder(
+        internal fun createBsonDecoder(
```

Comment:
Shouldn't this be `public` since [createBsonEncoder](https://github.com/mongodb/mongo-java-driver/pull/1510/files#diff-6ca11d6fb183c8e6441055f9fe4a6a804ae6676b009f6b6ba567b3d5fc757cbbR56) is public ?

Reply:
Thanks good catch - I opted to add an internal `BsonEncoderDecoderUtils` object. So that the public interfaces are as simple as can be and don't leak a useless companion object (from outside the library).